graphstorm-processing/graphstorm_processing/distributed_executor.py

```diff
@@ -652,7 +658,15 @@ def parse_args() -> argparse.Namespace:
             "data on the Spark leader if necessary."
         ),
     )
-
+    parser.add_argument(
+        "--num-threads",
```

Comment:
Let's change the name to --num-s3-threads or a better name.
`--num-threads` is vague.